src/Symfony/Component/Validator/Validation.php

```diff
@@ -23,10 +23,21 @@ final class Validation
 {
     /**
      * Creates a callable chain of constraints.
+     *
+     * @param Constraint|ValidatorInterface|null $constraintOrValidator
      */
-    public static function createCallable(Constraint ...$constraints): callable
+    public static function createCallable($constraintOrValidator = null, Constraint ...$constraints): callable
     {
-        $validator = self::createValidator();
+        $validator = $constraintOrValidator;
+
+        if ($constraintOrValidator instanceof Constraint) {
+            $constraints = \func_get_args();
+            $validator = null;
+        } elseif (null !== $constraintOrValidator && !$constraintOrValidator instanceof ValidatorInterface) {
+            throw new \TypeError(sprintf('Argument 1 passed to "%s()" must be a "%s" or a "%s" object, "%s" given.', __METHOD__, Constraint::class, ValidatorInterface::class, get_debug_type($constraintOrValidator)));
```

Comment:
The error message should say `or "null"` too.

Reply:
no need to advertise this case IMHO